### PR TITLE
Display NoResultPanel when no result are returned for query

### PR DESCRIPTION
**Pull request by G-Ray**
## Description
Display no result message in a panel when there is not result for current query (from search input or url query param)

## Screenshots
desktop

mobile gif

### CODE REVIEW

**Comment by bbecquet** [comment]
Can you put a screenshot of the resulting panel in the desc please?

File: src/panel/category/CategoryPanelError.jsx [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-      {zoomIn
-        ? _('Please zoom in the map to see the results for this category.', 'categories')
-        : _('No results found. Please zoom out of the map.', 'categories')
-      }
-    </p>
-    <button onClick={geoloc}>
-      <span className="icon-pin_geoloc" /> {_('Search around my position', 'categories')}
-    </button>
-  </div>;
+  return (
```
Comment (bbecquet) [inline src/panel/category/CategoryPanelError.jsx:11]: Not a fan of putting this unrelated change in the same PR.

File: src/scss/includes/panels/categories.scss [hunk 2]
```diff
@@ -38,14 +38,6 @@
 
 .category__panel__error {
   padding: 30px 40px 50px;
```
Comment (bbecquet) [inline src/scss/includes/panels/categories.scss:40]: If you want to go on with this change for visual consistency, let's set the bottom padding to the same value as the other panel…

File: src/panel/NoResultPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      white
+      close={close}
+    >
+      <Flex
+        flexDirection="column"
+        alignItems="center"
+        justifyContent="center"
+        style={{ padding: '0 16px 32px 16px' }}
+      >
+        <span className="u-mb-8 u-text--heading6">{_('No results found.')}</span>
```
Comment (bbecquet) [inline src/panel/NoResultPanel.jsx:25]: I would add some more margin here, the two elements are really close.

**Review by bbecquet — COMMENTED** [review]
Is this design Panel also validated on mobile?

File: src/components/ui/Flex.jsx [hunk 4]
```diff
@@ -8,6 +8,7 @@ const Flex = forwardRef(({
   className,
   justifyContent,
   alignItems = 'center',
+  flexDirection,
```
Comment (bbecquet) [inline src/components/ui/Flex.jsx:11]: Changes on this file aren't required anymore by the rest of the diff, they can be removed.

File: src/panel/NoResultPanel.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import React from 'react';
+
+import { Panel } from 'src/components/ui';
+
+const close = () => window.app.navigateTo('/');
+
+const handleSearchClick = e => {
+  e.preventDefault();
+  close();
+  window.__searchInput.searchInputHandle.value = '';
```
Comment (bbecquet) [inline src/panel/NoResultPanel.jsx:11]: To be tested, but you should be able to avoid this global stuff manip by importing `SearchInput` and calling `setInputValue` then `select`.

Comment (G-Ray) [reply src/panel/NoResultPanel.jsx]: I confirm it works! thansk

**Review by bbecquet — APPROVED** [review]
